otcextensions/sdk/modelartsv1/v1/devenv.py

```diff
@@ -0,0 +1,364 @@
+    )
+    #: Cause for a creation or startup failure.
+    failed_reasons = resource.Body("failed_reasons", type=FailedReasonsSpec)
+    #: Path for storing custom image logs.
+    log_path = resource.Body("log_path")
+    #: Git repository information.
+    repository = resource.Body("repository", type=RepositorySpec)
+    #: Time when the resource is reserved.
+    resource_reserved_timestamp = resource.Body(
+        "resource_reserved_timestamp", type=int
+    )
+    #: Storage path.
+    storage = resource.Body("storage", type=StorageSpec)
+
+
+class PoolSpec(resource.Resource):
+    #: ID of a resource pool.
+    id = resource.Body("id")
+    #: Name of a resource pool.
+    name = resource.Body("name")
+    #: This parameter is mandatory when type is set to USER_DEFINED.
+    owner = resource.Body("owner", type=dict)
+    #: Type of a resource pool.
+    type = resource.Body("type")
+
+
+class FlavorDetailsSpec(resource.Resource):
+    #: Billing specifications.
+    billing_flavor = resource.Body("billing_flavor")
+    #: Billing ratio This parameter is mandatory when billing_flavor is
+    #:  specified.
+    billing_params = resource.Body("billing_params", type=int)
+    #: Auto stop time after startup, in seconds.
+    duration = resource.Body("duration", type=int)
+    #: Number of instances of this flavor the current created.
+    instance_num = resource.Body("instance_num", type=int)
+    #: Whether the current user has the permission to use this flavor.
+    is_permitted = resource.Body("is_permitted", type=bool)
+    #: Flavor name.
+    name = resource.Body("name")
+    #: Parameters that describing flavor.
+    params = resource.Body("params", type=dict)
+    #: Parameters that describing flavor.
+    params_extends = resource.Body("params_extends", type=dict)
+    #: Promotion type.
+    promo_type = resource.Body("promo_type")
+    #: Left queuing time, in secondsThis parameter is mandatory when
+    #:  promo_type is set to Free and status is set to soldOut.
+    queue_left_time = resource.Body("queue_left_time", type=int)
+    #: This parameter is mandatory when promo_type is set to Free and status
+    #:  is set to soldOut.
+    queuing_num = resource.Body("queuing_num", type=int)
+    #: Flavor sale status The options are as follows:onSalesoldOut.
+    status = resource.Body("status")
+    #: Supported storage type.
+    storage_list = resource.Body("storage_list", type=list)
+    #: Maximum retention period of an inactive instance of this flavor in the
+    #:  database, in hoursThe default value is -1, indicating that the instance
+    #:  can be permanently saved.
+    store_time = resource.Body("store_time", type=int)
+    #: Flavor status.
+    type = resource.Body("type")
+
+
+class QueuingInfoSpec(resource.Resource):
+    #: Time when an instance starts queuing.
+    begin_timestamp = resource.Body("begin_timestamp", type=int)
+    #: Development environment type.
+    de_type = resource.Body("de_type")
```

Comment:
at least dev_env_type

Reply:
@vineet-pruthi don't spare the symbols, please create more readable variables

Reply:
updated to dev_env_type